Add compatibility for other React, Next.js, and compiled web frameworks

# Add compatibility for other React, Next.js, and compiled web frameworks
Currently, Aksharamukha doesn't work correctly unless imported via a script tag. In the Next.js documentation, for instance, it fails with the following error:
```
Cannot find module 'fs/promises'
```

This might be a limitation brought by Pyodide itself. Regardless, this needs to be researched and tackled.

--- a/.gitignore
+++ b/.gitignore
@@ -147,3 +147,6 @@ Thumbs.db
 
 # Wheel
 /downloads/
+
+# IDE
+.vscode/

--- a/docs/next.config.ts
+++ b/docs/next.config.ts
@@ -3,6 +3,7 @@ import type { NextConfig } from "next";
 const nextConfig: NextConfig = {
   /* config options here */
   output: "export",
+  outputFileTracingRoot: __dirname,
 };
 
 export default nextConfig;

--- a/docs/package.json
+++ b/docs/package.json
@@ -12,6 +12,7 @@
   "dependencies": {
     "@radix-ui/react-select": "^2.2.6",
     "@radix-ui/react-slot": "^1.2.3",
+    "aksharamukha": "file:..",
     "class-variance-authority": "^0.7.1",
     "clsx": "^2.1.1",
     "lucide-react": "^0.544.0",

--- a/docs/src/app/page.tsx
+++ b/docs/src/app/page.tsx
@@ -1,6 +1,5 @@
 "use client";
 
-import { useState, useEffect } from "react";
 import HeroSection from "@/components/HeroSection";
 import AboutSection from "@/components/AboutSection";
 import ExamplesSection from "@/components/ExamplesSection";
@@ -8,36 +7,7 @@ import LiveDemoSection from "@/components/LiveDemoSection";
 import QuickStartSection from "@/components/QuickStartSection";
 import Footer from "@/components/Footer";
 
-declare global {
-	interface Window {
-		// eslint-disable-next-line @typescript-eslint/no-explicit-any
-		Aksharamukha?: any;
-	}
-}
-
 export default function AksharamukhaPage() {
-	const [isLoaded, setIsLoaded] = useState(false);
-
-	useEffect(() => {
-		const script = document.createElement("script");
-		script.src =
-			"https://cdn.jsdelivr.net/npm/aksharamukha@latest/dist/index.global.js";
-		script.async = true;
-		script.onload = () => {
-			window.Aksharamukha.new()
-			// eslint-disable-next-line @typescript-eslint/no-explicit-any
-			.then((aksharamukha: any) => {
-				window.aksharamukha = aksharamukha;
-				setIsLoaded(true);
-			});
-		};
-		document.body.appendChild(script);
-
-		return () => {
-			document.body.removeChild(script);
-		};
-	}, []);
-
 	const scrollToDemo = () => {
 		document.getElementById("demo")?.scrollIntoView({ behavior: "smooth" });
 	};
@@ -46,8 +16,8 @@ export default function AksharamukhaPage() {
 		<div>
 			<HeroSection onGetStartedClick={scrollToDemo} />
 			<AboutSection />
-			<ExamplesSection isLoaded={isLoaded} />
-			<LiveDemoSection isLoaded={isLoaded} />
+			<ExamplesSection/>
+			<LiveDemoSection/>
 			<QuickStartSection />
 			<Footer />
 		</div>

--- a/docs/src/components/ExamplesSection.tsx
+++ b/docs/src/components/ExamplesSection.tsx
@@ -1,6 +1,6 @@
 import ScriptShowcase from "./ScriptShowcase";
 
-export default function ExamplesSection({ isLoaded }: { isLoaded: boolean }) {
+export default function ExamplesSection() {
 	return (
 		<section className="min-h-screen flex items-center justify-center border-b border-gray-200/30 dark:border-gray-700/30 bg-gradient-to-b from-background to-teal-50/30 dark:to-teal-950/10 py-8 sm:py-16">
 			<div className="container mx-auto px-4 w-full">
@@ -53,7 +53,7 @@ export default function ExamplesSection({ isLoaded }: { isLoaded: boolean }) {
 							</div>
 						</div>
 					</div>
-					<ScriptShowcase isLoaded={isLoaded} />
+					<ScriptShowcase/>
 				</div>
 			</div>
 		</section>

--- a/docs/src/components/LiveDemoSection.tsx
+++ b/docs/src/components/LiveDemoSection.tsx
@@ -1,4 +1,6 @@
-import { useState, useEffect, useCallback } from "react";
+import { useState, useEffect, useCallback, useMemo } from "react";
+import Aksharamukha, { Script, Scripts } from "aksharamukha";
+
 import { Button } from "@/components/ui/button";
 import { Card } from "@/components/ui/card";
 import { Textarea } from "@/components/ui/textarea";
@@ -13,28 +15,22 @@ import { Copy, Check, Shuffle, ArrowRightLeft } from "lucide-react";
 
 declare global {
 	interface Window {
-		aksharamukha?: {
-			process: (
-				text: string,
-				fromScript: string,
-				toScript: string
-			) => Promise<string>;
-		};
+		aksharamukha?: Aksharamukha;
 	}
 }
 
 const scripts = [
-	{ code: 'kaithi', value: "Kaithi", label: "𑂍𑂶𑂟𑂲 (Kaithi)", example: "𑂣𑂹𑂩𑂝𑂰𑂧𑂹" },
-	{ code: 'devanagari', value: "Devanagari", label: "देवनागरी (Devanagari)", example: "प्रणाम्" },
-	{ code: 'tamil', value: "Tamil", label: "தமிழ் (Tamil)", example: "வணக்கம்" },
-	{ code: 'kannada', value: "Kannada", label: "ಕನ್ನಡ (Kannada)", example: "ನಮಸ್ಕಾರ" },
-	{ code: 'telugu', value: "Telugu", label: "తెలుగు (Telugu)", example: "నమస్కారం" },
-	{ code: 'malayalam', value: "Malayalam", label: "മലയാളം (Malayalam)", example: "നമസ്കാരം" },
-	{ code: 'bengali', value: "Bengali", label: "বাংলা (Bengali)", example: "নমস্কার" },
-	{ code: 'gujarati', value: "Gujarati", label: "ગુજરાતી (Gujarati)", example: "નમસ્કાર" },
-	{ code: 'gurmukhi', value: "Gurmukhi", label: "ਗੁਰਮੁਖੀ (Gurmukhi)", example: "ਸਤ ਸ੍ਰੀ ਅਕਾਲ" },
-	{ code: 'oriya', value: "Oriya", label: "ଓଡ଼ିଆ (Oriya)", example: "ନମସ୍କାର" },
-	{ code: 'itrans', value: "ITRANS", label: "Latin (ITRANS)", example: "namaskara" },
+	{ code: Scripts.Kaithi, label: "𑂍𑂶𑂟𑂲 (Kaithi)", example: "𑂣𑂹𑂩𑂝𑂰𑂧𑂹" },
+	{ code: Scripts.Devanagari, label: "देवनागरी (Devanagari)", example: "प्रणाम्" },
+	{ code: Scripts.Tamil, label: "தமிழ் (Tamil)", example: "வணக்கம்" },
+	{ code: Scripts.Kannada, label: "ಕನ್ನಡ (Kannada)", example: "ನಮಸ್ಕಾರ" },
+	{ code: Scripts.Telugu, label: "తెలుగు (Telugu)", example: "నమస్కారం" },
+	{ code: Scripts.Malayalam, label: "മലയാളം (Malayalam)", example: "നമസ്കാരം" },
+	{ code: Scripts.BengaliBangla, label: "বাংলা (Bengali)", example: "নমস্কার" },
+	{ code: Scripts.Gujarati, label: "ગુજરાતી (Gujarati)", example: "નમસ્કાર" },
+	{ code: Scripts.PunjabiGurmukhi, label: "ਗੁਰਮੁਖੀ (Gurmukhi)", example: "ਸਤ ਸ੍ਰੀ ਅਕਾਲ" },
+	{ code: Scripts.OriyaOdia, label: "ଓଡ଼ିଆ (Oriya)", example: "ନମସ୍କାର" },
+	{ code: Scripts.ITRANS, label: "Latin (ITRANS)", example: "namaskara" },
 ];
 
 const sampleTexts = [
@@ -46,17 +42,25 @@ const sampleTexts = [
 	"kShamaa karie",
 ];
 
-interface LiveDemoSectionProps {
-	isLoaded: boolean;
-}
-
-export default function LiveDemoSection({ isLoaded }: LiveDemoSectionProps) {
+export default function LiveDemoSection() {
+	const [client, setClient] = useState<Aksharamukha | null>(null);
 	const [inputText, setInputText] = useState("praNAm.");
 	const [outputText, setOutputText] = useState("");
-	const [fromScript, setFromScript] = useState("itrans");
-	const [toScript, setToScript] = useState("kaithi");
+	const [fromScript, setFromScript] = useState<Script>(Scripts.ITRANS);
+	const [toScript, setToScript] = useState<Script>(Scripts.Kaithi);
 	const [copied, setCopied] = useState(false);
 	const [isConverting, setIsConverting] = useState(false);
+	
+	const isLoaded = useMemo(() => {
+		return client != null;
+	}, [client]);
+
+	useEffect(() => {
+		(async function() {
+			const client = await Aksharamukha.new();
+			setClient(client);
+		})()
+	}, [setClient]);
 
 	const swapScripts = useCallback(() => {
 		const temp = fromScript;
@@ -72,11 +76,16 @@ export default function LiveDemoSection({ isLoaded }: LiveDemoSectionProps) {
 	}, []);
 
 	useEffect(() => {
-		if (isLoaded && window.aksharamukha && inputText) {
+		if (isLoaded && inputText) {
+			if (!client) {
+				alert("Aksharamukha failed to load. Please refresh the page, and if the issue persists, report it on GitHub.");
+				return;
+			}
+
 			setIsConverting(true);
 			const timeoutId = setTimeout(async () => {
 				try {
-					const result = await window.aksharamukha?.process(
+					const result = client?.process(
 						fromScript,
 						toScript,
 						inputText
@@ -91,7 +100,7 @@ export default function LiveDemoSection({ isLoaded }: LiveDemoSectionProps) {
 
 			return () => clearTimeout(timeoutId);
 		}
-	}, [inputText, fromScript, toScript, isLoaded]);
+	}, [inputText, fromScript, toScript, isLoaded, client]);
 
 	const handleCopy = async () => {
 		await navigator.clipboard.writeText(outputText);
@@ -166,7 +175,7 @@ export default function LiveDemoSection({ isLoaded }: LiveDemoSectionProps) {
 									</label>
 									<Select
 										value={fromScript}
-										onValueChange={setFromScript}
+										onValueChange={value => setFromScript(value as Script)}
 									>
 										<SelectTrigger className="border-teal-200/60 dark:border-teal-800/50 focus:border-teal-400 dark:focus:border-teal-600 focus:ring-teal-500/20 hover:border-teal-300/70 dark:hover:border-teal-700/60 transition-all duration-200">
 											<SelectValue />
@@ -174,7 +183,7 @@ export default function LiveDemoSection({ isLoaded }: LiveDemoSectionProps) {
 										<SelectContent className="border-teal-200/60 dark:border-teal-800/50">
 											{scripts.map((script) => (
 												<SelectItem
-													key={script.value}
+													key={script.code}
 													value={script.code}
 													className="hover:bg-teal-50/80 dark:hover:bg-teal-950/50"
 												>
@@ -234,7 +243,7 @@ export default function LiveDemoSection({ isLoaded }: LiveDemoSectionProps) {
 									</label>
 									<Select
 										value={toScript}
-										onValueChange={setToScript}
+										onValueChange={value => setToScript(value as Script)}
 									>
 										<SelectTrigger className="border-teal-200/60 dark:border-teal-800/50 focus:border-teal-400 dark:focus:border-teal-600 focus:ring-teal-500/20 hover:border-teal-300/70 dark:hover:border-teal-700/60 transition-all duration-200">
 											<SelectValue />
@@ -242,7 +251,7 @@ export default function LiveDemoSection({ isLoaded }: LiveDemoSectionProps) {
 										<SelectContent className="border-teal-200/60 dark:border-teal-800/50">
 											{scripts.map((script) => (
 												<SelectItem
-													key={script.value}
+													key={script.code}
 													value={script.code}
 													className="hover:bg-teal-50/80 dark:hover:bg-teal-950/50"
 												>

--- a/docs/src/components/ScriptShowcase.tsx
+++ b/docs/src/components/ScriptShowcase.tsx
@@ -1,16 +1,11 @@
 "use client";
 
-import React, { useEffect, useState } from "react";
+import React, { useEffect, useMemo, useState } from "react";
+import Aksharamukha, { Script, Scripts } from "aksharamukha";
 
 declare global {
 	interface Window {
-		aksharamukha?: {
-			process: (
-				from: string,
-				to: string,
-				text: string
-			) => Promise<string>;
-		};
+		aksharamukha?: Aksharamukha;
 	}
 }
 
@@ -18,8 +13,8 @@ type Example = {
 	label: string; // short label like "ITRANS → Telugu"
 	fromScript: string;
 	toScript: string;
-	from: string;
-	to: string;
+	from: Script;
+	to: Script;
 	source: string;
 	expected: string;
 };
@@ -29,8 +24,8 @@ const EXAMPLES: Example[] = [
 		label: "ITRANS → Telugu",
 		fromScript: "ITRANS",
 		toScript: "Telugu",
-		from: 'itrans',
-		to: 'telugu',
+		from: Scripts.ITRANS,
+		to: Scripts.Telugu,
 		source: "rAma",
 		expected: "రామ",
 	},
@@ -38,8 +33,8 @@ const EXAMPLES: Example[] = [
 		label: "Harvard-Kyoto → Siddham",
 		fromScript: "Harvard-Kyoto",
 		toScript: "Siddham",
-		from: 'hk',
-		to: 'siddham',
+		from: Scripts.HarvardKyoto,
+		to: Scripts.Siddham,
 		source: "buddhaH",
 		expected: "𑖤𑖲𑖟𑖿𑖠𑖾",
 	},
@@ -47,14 +42,15 @@ const EXAMPLES: Example[] = [
 		label: "Devanagari → Malayalam",
 		fromScript: "Devanagari",
 		toScript: "Malayalam",
-		from: 'devanagari',
-		to: 'malayalam',
+		from: Scripts.Devanagari,
+		to: Scripts.Malayalam,
 		source: "धर्म",
 		expected: "ധര്മ",
 	},
 ];
 
-export default function ScriptShowcase({ isLoaded }: { isLoaded: boolean }) {
+export default function ScriptShowcase() {
+	const [client, setClient] = useState<Aksharamukha | null>(null);
 	const [index, setIndex] = useState<number>(0);
 	const [firstConversion, setFirstConversion] = useState<boolean>(false);
 	const [converted, setConverted] = useState<string>(EXAMPLES[0].expected);
@@ -62,19 +58,35 @@ export default function ScriptShowcase({ isLoaded }: { isLoaded: boolean }) {
 	const [inError, setInError] = useState<boolean>(false);
 	const cycleInterval = 3000;
 
+	const isLoaded = useMemo(() => {
+		return client != null;
+	}, [client]);
+
+	useEffect(() => {
+		(async function() {
+			const client = await Aksharamukha.new();
+			setClient(client);
+		})()
+	}, [setClient]);
+
 	// perform conversion for the current example
 	useEffect(() => {
 		const ex = EXAMPLES[index];
 		setConverted(ex.expected);
 		setLoading(true);
 
+		if (!isLoaded) {
+			return;
+		}
+
 		async function doConvert() {
-			if (!isLoaded || !window?.aksharamukha?.process) {
+			if (!client) {
+				alert("Aksharamukha failed to load. Please refresh the page, and if the issue persists, report it on GitHub.");
 				return;
 			}
 			
 			try {
-				const out = await window.aksharamukha!.process(
+				const out = client.process(
 					ex.from,
 					ex.to,
 					ex.source
@@ -93,10 +105,7 @@ export default function ScriptShowcase({ isLoaded }: { isLoaded: boolean }) {
 
 		doConvert();
 		return () => {};
-	}, [
-		index,
-		isLoaded
-	]);
+	}, [index, isLoaded, client]);
 
 	useEffect(() => {
 		if (!firstConversion) {

--- a/package.json
+++ b/package.json
@@ -8,9 +8,12 @@
 	"types": "dist/index.d.ts",
 	"exports": {
 		".": {
-			"browser": "./dist/index.global.js",
+			"types": "./dist/index.d.ts",
+			"browser": "./dist/index.js",
+			"node": "./dist/index.js",
+			"import": "./dist/index.js",
 			"require": "./dist/index.cjs",
-			"import": "./dist/index.js"
+			"default": "./dist/index.js"
 		}
 	},
 	"scripts": {

--- a/src/aksharamukha/index.test.ts
+++ b/src/aksharamukha/index.test.ts
@@ -15,6 +15,11 @@ describe('Aksharamukha', () => {
 
 		beforeAll(async () => {
 			Aksharamukha._isTestEnv = true;
+			Aksharamukha._testLoadPyodide = async () => await loadPyodide({
+				indexURL: './node_modules/pyodide',
+				packageCacheDir: './node_modules/pyodide'
+			});
+			Aksharamukha._testFS = await import('fs');
 		});
 
 		afterAll(() => {

--- a/src/aksharamukha/index.ts
+++ b/src/aksharamukha/index.ts
@@ -1,5 +1,5 @@
 import { type PyodideInterface } from 'pyodide';
-import { aksharamukhaCDNIndexURL, /* pyodideCDNIndexURL, pyodideCDNModuleURL, */ wheels } from '../constants';
+import { aksharamukhaCDNIndexURL, wheels } from '../constants';
 import { fixPostOptions, PostOption, PreOption, Script, Scripts } from '../enums';
 
 const isNode =
@@ -54,7 +54,6 @@ async function getLoadPyodide() {
 
 		const pyodideModule = isNode
 			? await dynamicImport('pyodide')
-			// : await dynamicImport(pyodideCDNModuleURL);
 			: await dynamicImport(`${aksharamukhaCDNIndexURL}/pyodide.mjs`);
 		loadPyodideRef = pyodideModule.loadPyodide;
 	}
@@ -63,9 +62,12 @@ async function getLoadPyodide() {
 }
 
 export default class Aksharamukha {
-	static _isTestEnv: boolean = false;
 	static _currentScript: HTMLScriptElement;
 	pyodide: PyodideInterface;
+
+	static _isTestEnv: boolean = false;
+	static _testLoadPyodide: typeof import('pyodide')['loadPyodide'];
+	static _testFS: typeof import('fs');
 
 	private constructor(pyodide: PyodideInterface) {
 		this.pyodide = pyodide;
@@ -79,7 +81,10 @@ export default class Aksharamukha {
 		let pyodide = opts?.pyodide;
 		if (pyodide == null) {
 			if (this._isTestEnv) {
-				pyodide = await loadTestPyodide();
+				if (this._testLoadPyodide == null) {
+					throw new Error('Test Pyodide instance not set. Please set Aksharamukha._testLoadPyodide before running tests.');
+				}
+				pyodide = await this._testLoadPyodide();
 			} else {
 				const loadPyodide = await getLoadPyodide();
 				pyodide = isNode
@@ -91,8 +96,16 @@ export default class Aksharamukha {
 		const micropip = await ensureMicropip(pyodide);
 
 		if (isNode) {
-			// Keep import target non-literal so browser bundlers don't object to the import of "fs" in this Node-only block.
-			const fs = await (new Function('specifier', 'return import(specifier)') as (specifier: string) => Promise<typeof import('fs')>)('fs');
+			let fs: typeof import('fs');
+			if (this._isTestEnv) {
+				if (this._testFS == null) {
+					throw new Error('Test fs module not set. Please set Aksharamukha._testFS before running tests.');
+				}
+				fs = this._testFS;
+			} else {
+				// Keep import target non-literal so browser bundlers don't object to the import of "fs" in this Node-only block.
+				fs = await (new Function('specifier', 'return import(specifier)') as (specifier: string) => Promise<typeof import('fs')>)('fs');
+			}
 
 			for (const wheel of wheels) {
 				let wheelData: Buffer<ArrayBuffer>;
@@ -206,14 +219,6 @@ export default class Aksharamukha {
 		const cmd = buildAutoDetectCMD({ txt, plugin });
 		return this.pyodide.runPythonAsync(cmd);
 	}
-}
-
-async function loadTestPyodide(): Promise<PyodideInterface> {
-	const loadPyodide = await getLoadPyodide();
-	return await loadPyodide({
-		indexURL: './node_modules/pyodide',
-		packageCacheDir: './node_modules/pyodide'
-	})
 }
 
 async function ensureMicropip(pyodide: PyodideInterface) {

--- a/src/aksharamukha/index.ts
+++ b/src/aksharamukha/index.ts
@@ -46,22 +46,8 @@ export type AksharamukhaInitOptions = {
 	pyodide?: PyodideInterface;
 };
 
-let loadPyodideRef: typeof import('pyodide')['loadPyodide'] | undefined;
-
-async function getLoadPyodide() {
-	if (loadPyodideRef == null) {
-		const dynamicImport = new Function('specifier', 'return import(specifier)') as (specifier: string) => Promise<{ loadPyodide: typeof import('pyodide')['loadPyodide'] }>;
-
-		const pyodideModule = isNode
-			? await dynamicImport('pyodide')
-			: await dynamicImport(`${aksharamukhaCDNIndexURL}/pyodide.mjs`);
-		loadPyodideRef = pyodideModule.loadPyodide;
-	}
-
-	return loadPyodideRef;
-}
-
 export default class Aksharamukha {
+	static _loadPyodideRef: typeof import('pyodide')['loadPyodide'] | undefined;
 	static _currentScript: HTMLScriptElement;
 	pyodide: PyodideInterface;
 
@@ -77,6 +63,38 @@ export default class Aksharamukha {
 		this._currentScript = script;
 	}
 
+	private static getCurrentScriptPath(): string {
+		if (this._currentScript) {
+			const scriptPath = this._currentScript.src;
+			const parentPath = scriptPath.substring(0, scriptPath.lastIndexOf('/'));
+			return parentPath;
+		}
+
+		throw new Error('Could not determine script path for browser environment. Ensure that the script is properly included in the HTML with a src attribute.');
+	}
+
+	private static async getLoadPyodide() {
+		if (this._loadPyodideRef == null) {
+			const dynamicImport = new Function('specifier', 'return import(specifier)') as (specifier: string) => Promise<typeof import('pyodide')>;
+			// const dynamicImport = new Function('specifier', 'return import(specifier)') as (specifier: string) => Promise<{ loadPyodide: typeof import('pyodide')['loadPyodide'] }>;
+
+			let pyodideModule: typeof import('pyodide');
+			if (isNode) {
+				pyodideModule = await dynamicImport('pyodide');
+			} else {
+				try {
+					const parentPath = this.getCurrentScriptPath();
+					pyodideModule = await dynamicImport(`${parentPath}/pyodide.mjs`);
+				} catch {
+					pyodideModule = await dynamicImport(`${aksharamukhaCDNIndexURL}/pyodide.mjs`);
+				}
+			}
+			this._loadPyodideRef = pyodideModule.loadPyodide;
+		}
+
+		return this._loadPyodideRef;
+	}
+
 	public static async new(opts?: AksharamukhaInitOptions): Promise<Aksharamukha> {
 		let pyodide = opts?.pyodide;
 		if (pyodide == null) {
@@ -86,10 +104,16 @@ export default class Aksharamukha {
 				}
 				pyodide = await this._testLoadPyodide();
 			} else {
-				const loadPyodide = await getLoadPyodide();
-				pyodide = isNode
-					? await loadPyodide()
-					: await loadPyodide({ indexURL: aksharamukhaCDNIndexURL });
+				const loadPyodide = await this.getLoadPyodide();
+				try {
+					pyodide = await loadPyodide({ indexURL: this.getCurrentScriptPath() });
+				} catch {
+					try {
+						pyodide = await loadPyodide();
+					} catch {
+						pyodide = await loadPyodide({ indexURL: aksharamukhaCDNIndexURL });
+					}
+				}
 			}
 		}
 
@@ -130,8 +154,7 @@ export default class Aksharamukha {
 			}
 		} else {
 			try {
-				const scriptPath = this._currentScript.src;
-				const parentPath = scriptPath.substring(0, scriptPath.lastIndexOf('/'));
+				const parentPath = this.getCurrentScriptPath();
 				await micropip.install(wheels.map(wheel => `${parentPath}/${wheel}`), { keep_going: true })
 			} catch {
 				await micropip.install(wheels.map(wheel => `${aksharamukhaCDNIndexURL}/${wheel}`), { keep_going: true })

--- a/src/aksharamukha/index.ts
+++ b/src/aksharamukha/index.ts
@@ -76,7 +76,6 @@ export default class Aksharamukha {
 	private static async getLoadPyodide() {
 		if (this._loadPyodideRef == null) {
 			const dynamicImport = new Function('specifier', 'return import(specifier)') as (specifier: string) => Promise<typeof import('pyodide')>;
-			// const dynamicImport = new Function('specifier', 'return import(specifier)') as (specifier: string) => Promise<{ loadPyodide: typeof import('pyodide')['loadPyodide'] }>;
 
 			let pyodideModule: typeof import('pyodide');
 			if (isNode) {

--- a/src/aksharamukha/index.ts
+++ b/src/aksharamukha/index.ts
@@ -1,8 +1,11 @@
-import { loadPyodide, type PyodideInterface } from 'pyodide';
-import { wheelBaseURL, wheels } from '../constants';
+import { type PyodideInterface } from 'pyodide';
+import { aksharamukhaCDNIndexURL, /* pyodideCDNIndexURL, pyodideCDNModuleURL, */ wheels } from '../constants';
 import { fixPostOptions, PostOption, PreOption, Script, Scripts } from '../enums';
 
-const isNode = typeof window === 'undefined' || (typeof process !== 'undefined' && process.versions?.node);
+const isNode =
+	typeof globalThis !== 'undefined' &&
+	typeof (globalThis as { process?: unknown }).process === 'object' &&
+	typeof (globalThis as { process?: { versions?: { node?: unknown } } }).process?.versions?.node === 'string';
 
 export type ProcessArgs = {
 	src: Script,
@@ -43,6 +46,22 @@ export type AksharamukhaInitOptions = {
 	pyodide?: PyodideInterface;
 };
 
+let loadPyodideRef: typeof import('pyodide')['loadPyodide'] | undefined;
+
+async function getLoadPyodide() {
+	if (loadPyodideRef == null) {
+		const dynamicImport = new Function('specifier', 'return import(specifier)') as (specifier: string) => Promise<{ loadPyodide: typeof import('pyodide')['loadPyodide'] }>;
+
+		const pyodideModule = isNode
+			? await dynamicImport('pyodide')
+			// : await dynamicImport(pyodideCDNModuleURL);
+			: await dynamicImport(`${aksharamukhaCDNIndexURL}/pyodide.mjs`);
+		loadPyodideRef = pyodideModule.loadPyodide;
+	}
+
+	return loadPyodideRef;
+}
+
 export default class Aksharamukha {
 	static _isTestEnv: boolean = false;
 	static _currentScript: HTMLScriptElement;
@@ -62,15 +81,18 @@ export default class Aksharamukha {
 			if (this._isTestEnv) {
 				pyodide = await loadTestPyodide();
 			} else {
-				pyodide = await loadPyodide();
+				const loadPyodide = await getLoadPyodide();
+				pyodide = isNode
+					? await loadPyodide()
+					: await loadPyodide({ indexURL: aksharamukhaCDNIndexURL });
 			}
 		}
 
-		await pyodide.loadPackage('micropip');
-		const micropip = pyodide.pyimport('micropip');
+		const micropip = await ensureMicropip(pyodide);
 
 		if (isNode) {
-			const fs = await import('fs');
+			// Keep import target non-literal so browser bundlers don't object to the import of "fs" in this Node-only block.
+			const fs = await (new Function('specifier', 'return import(specifier)') as (specifier: string) => Promise<typeof import('fs')>)('fs');
 
 			for (const wheel of wheels) {
 				let wheelData: Buffer<ArrayBuffer>;
@@ -99,7 +121,7 @@ export default class Aksharamukha {
 				const parentPath = scriptPath.substring(0, scriptPath.lastIndexOf('/'));
 				await micropip.install(wheels.map(wheel => `${parentPath}/${wheel}`), { keep_going: true })
 			} catch {
-				await micropip.install(wheels.map(wheel => `${wheelBaseURL}/${wheel}`), { keep_going: true })
+				await micropip.install(wheels.map(wheel => `${aksharamukhaCDNIndexURL}/${wheel}`), { keep_going: true })
 			}
 		}
 
@@ -187,10 +209,21 @@ export default class Aksharamukha {
 }
 
 async function loadTestPyodide(): Promise<PyodideInterface> {
+	const loadPyodide = await getLoadPyodide();
 	return await loadPyodide({
 		indexURL: './node_modules/pyodide',
 		packageCacheDir: './node_modules/pyodide'
 	})
+}
+
+async function ensureMicropip(pyodide: PyodideInterface) {
+	await pyodide.loadPackage('micropip');
+	try {
+		await pyodide.runPythonAsync('import micropip');
+		return pyodide.pyimport('micropip');
+	} catch (error) {
+		throw new Error(`Failed to initialize micropip in Pyodide. ${String(error)}`);
+	}
 }
 
 function buildProcessCMD(props: ProcessArgs) {

--- a/src/constants.ts
+++ b/src/constants.ts
@@ -4,5 +4,6 @@ export const wheels = [
 	'aksharamukha-2.3-py3-none-any.whl'
 ];
 
+// TODO: Fix
 export const aksharamukhaCDNIndexURL = `https://cdn.jsdelivr.net/npm/aksharamukha@2.3.0-staging2.0/dist`;
 export const wheelBaseURL = 'https://github.com/paramsiddharth/aksharamukha.js/releases/download/v2.3.0-0';

--- a/src/constants.ts
+++ b/src/constants.ts
@@ -4,4 +4,9 @@ export const wheels = [
 	'aksharamukha-2.3-py3-none-any.whl'
 ];
 
+// declare const __PYODIDE_VERSION__: string;
+// export const pyodideCDNModuleURL = `https://cdn.jsdelivr.net/pyodide/v${__PYODIDE_VERSION__}/full/pyodide.mjs`;
+// export const pyodideCDNIndexURL = `https://cdn.jsdelivr.net/pyodide/v${__PYODIDE_VERSION__}/full/`;
+
+export const aksharamukhaCDNIndexURL = `https://cdn.jsdelivr.net/npm/aksharamukha@2.3.0-staging2.0/dist`;
 export const wheelBaseURL = 'https://github.com/paramsiddharth/aksharamukha.js/releases/download/v2.3.0-0';

--- a/src/constants.ts
+++ b/src/constants.ts
@@ -4,9 +4,5 @@ export const wheels = [
 	'aksharamukha-2.3-py3-none-any.whl'
 ];
 
-// declare const __PYODIDE_VERSION__: string;
-// export const pyodideCDNModuleURL = `https://cdn.jsdelivr.net/pyodide/v${__PYODIDE_VERSION__}/full/pyodide.mjs`;
-// export const pyodideCDNIndexURL = `https://cdn.jsdelivr.net/pyodide/v${__PYODIDE_VERSION__}/full/`;
-
 export const aksharamukhaCDNIndexURL = `https://cdn.jsdelivr.net/npm/aksharamukha@2.3.0-staging2.0/dist`;
 export const wheelBaseURL = 'https://github.com/paramsiddharth/aksharamukha.js/releases/download/v2.3.0-0';

--- a/tsup.config.ts
+++ b/tsup.config.ts
@@ -1,10 +1,19 @@
 import { defineConfig } from 'tsup';
 
+// // @ts-expect-error Runtime Node import; this TS setup doesn't resolve built-in module types here.
+// import { readFileSync } from 'fs';
+
 const pyodideDependencies = [
 	'pyodide-lock.json',
 	'pyodide.asm.wasm',
 	'pyodide.asm.js',
+	'pyodide.mjs',
 ];
+
+/* const packageJson = JSON.parse(readFileSync('./package.json', 'utf8')) as {
+	dependencies?: Record<string, string>;
+};
+const pyodideVersionFromDependency = packageJson.dependencies?.pyodide?.match(/\d+\.\d+\.\d+/)?.[0]; */
 
 export default defineConfig({
 	entry: ['src/index.ts'],
@@ -14,6 +23,9 @@ export default defineConfig({
 	sourcemap: true,
 	minify: true,
 	clean: true,
+	/* define: {
+		__PYODIDE_VERSION__: JSON.stringify(pyodideVersionFromDependency),
+	}, */
 	esbuildOptions: options => {
 		if (options.format !== 'esm') {
 			options.logOverride = {

--- a/tsup.config.ts
+++ b/tsup.config.ts
@@ -1,19 +1,11 @@
 import { defineConfig } from 'tsup';
 
-// // @ts-expect-error Runtime Node import; this TS setup doesn't resolve built-in module types here.
-// import { readFileSync } from 'fs';
-
 const pyodideDependencies = [
 	'pyodide-lock.json',
 	'pyodide.asm.wasm',
 	'pyodide.asm.js',
 	'pyodide.mjs',
 ];
-
-/* const packageJson = JSON.parse(readFileSync('./package.json', 'utf8')) as {
-	dependencies?: Record<string, string>;
-};
-const pyodideVersionFromDependency = packageJson.dependencies?.pyodide?.match(/\d+\.\d+\.\d+/)?.[0]; */
 
 export default defineConfig({
 	entry: ['src/index.ts'],
@@ -23,9 +15,6 @@ export default defineConfig({
 	sourcemap: true,
 	minify: true,
 	clean: true,
-	/* define: {
-		__PYODIDE_VERSION__: JSON.stringify(pyodideVersionFromDependency),
-	}, */
 	esbuildOptions: options => {
 		if (options.format !== 'esm') {
 			options.logOverride = {

